test/distributed/checkpoint/test_checkpoint.py

```diff
@@ -198,6 +198,10 @@ def write_data(
     def finish(self, metadata: Metadata, results: List[List[WriteResult]]) -> None:
         self._fail_rank("fail_finish")
 
+    @staticmethod
```

Comment:
nit: I tend to favor @classmethod in instances such as these since it has better OO support.